docfx/docs/Examples.md

```diff
@@ -185,3 +172,152 @@ static unsafe void Main(string[] args)
     psi->Release();
 }
 ```
+
+## Using COM classes (NetFwMgr, INetFwMgr)
+
+In this example we see how to activate a COM class and call methods on it via its primary interface. When marshalling
+is enabled, we can use C# cast to do the "QueryInterface" that you would have seen in a native C++ sample.
+
+```
+NetFwMgr
+INetFwMgr
+IEnumVARIANT
+INetFwAuthorizedApplication
+```
+
+### Marshalling enabled (built-in COM Interop, not AOT-compatible)
+
+```cs
+var fwMgr = (INetFwMgr)new NetFwMgr();
+var authorizedApplications = fwMgr.LocalPolicy.CurrentProfile.AuthorizedApplications;
+var aaObjects = new object[authorizedApplications.Count];
+var applicationsEnum = (IEnumVARIANT)authorizedApplications._NewEnum;
+applicationsEnum.Next((uint)authorizedApplications.Count, aaObjects, out uint fetched);
+foreach (var aaObject in aaObjects)
+{
+    var app = (INetFwAuthorizedApplication)aaObject;
+    Console.WriteLine("---");
+    Console.WriteLine($"Name: {app.Name.ToString()}");
+    Console.WriteLine($"Enabled: {(bool)app.Enabled}");
+    Console.WriteLine($"Remote Addresses: {app.RemoteAddresses.ToString()}");
+    Console.WriteLine($"Scope: {app.Scope}");
+    Console.WriteLine($"Process Image Filename: {app.ProcessImageFileName.ToString()}");
+    Console.WriteLine($"IP Version: {app.IpVersion}");
+}
+```
+
+### Marshalling enabled (COM wrappers, AOT compatible)
+
+Note that in COM wrappers mode, the generated interfaces have get_ methods instead of properties. Some parameters
+are also ComVariant instead of object because source generated COM does not support as much automatic marshalling
+as built-in COM does.
+
+```cs
+var fwMgr = NetFwMgr.CreateInstance<INetFwMgr>();
+var authorizedApplications = fwMgr.get_LocalPolicy().get_CurrentProfile().get_AuthorizedApplications();
+var aaObjects = new ComVariant[authorizedApplications.get_Count()];
+var applicationsEnum = (IEnumVARIANT)authorizedApplications.get__NewEnum();
+applicationsEnum.Next((uint)authorizedApplications.get_Count(), aaObjects, out uint fetched);
+foreach (var aaObject in aaObjects)
+{
+    var app = (INetFwAuthorizedApplication)ComVariantMarshaller.ConvertToManaged(aaObject)!;
+
+    Console.WriteLine("---");
+    Console.WriteLine($"Name: {app.get_Name().ToString()}");
+    Console.WriteLine($"Enabled: {(bool)app.get_Enabled()}");
+    Console.WriteLine($"Remote Addresses: {app.get_RemoteAddresses().ToString()}");
```

Comment:
Nit: no need to call `ToString()` within an interpolation hole, it's already called

Reply:
Removed.

Reply:
Please re-review! :)